python/essex-config/README.md

```diff
@@ -0,0 +1,195 @@
+# essex-configuration
+
+essex-configuration is a Python library for creating configuration objects that can read from various sources, including files, environment variables, and Azure Key Vault.
+
+## Installation
+
+To install the essex-configuration library, use pip:
+
+`pip install essex-configuration`
+
+## Basic Usage
+
+Here's how to create a configuration object for connecting to a customer database:
+
+
+```python
+from typing import Annotated, ClassVar
+from essex_config import Config, ConfigurationField, FieldVisibility
+
+
+class CustomerDatabase(Config):
+    """Class that holds the configuration to connect to the Customer Database"""
+
+    host: Annotated[str, ConfigurationField(
```

Comment:
So I like it to be Annotated because I feel like the ConfigurationField is metadata and not something you should "execute" the other reason I like it to be like this is because this way you can actually use Field() from pydantic (since Config is a BaseModel) you can do stuff like this:

```python
def test_basic_configuration():
    class BasicConfig(Config):
        __sources__: ClassVar[list[Source]] = [MockSource()]

        value1: str
        value2: int
        value3: float
        value4: bool
        value5: Annotated[str, ConfigurationField(description="value from Field()")] = Field(default_factory=lambda: "default5")

    config = BasicConfig.get_config()
    assert config.value1 == "value1"
    assert config.value2 == 2
    assert config.value3 == 3.0
    assert config.value4 is True
    assert config.value5 == "default5"
```

Super simple example but demonstrates the point haha. (Also as you can see now we can not use the Annotated[]...

Reply:
Would it be possible to make ConfigurationField inherit from Field? So it probably does not break the base model?